components/SearchBar.js

```diff
@@ -0,0 +1,81 @@
+import styled from "styled-components";
+import { useState, useEffect } from "react";
+import Fuse from "fuse.js";
+import Magnifier from "../public/magnify.svg";
+
+const StyledContainer = styled.div`
+  height: auto;
+  display: flex;
+  position: relative;
+`;
+
+const StyledMagnifier = styled(Magnifier)`
+  width: 1.8rem;
+  fill: var(--main-dark);
+  position: absolute;
+  top: calc(50% - 15px);
+  left: 3px;
+`;
+
+const StyledInput = styled.input`
+  width: ${({ $isShown }) => ($isShown ? "80vw" : "35px")};
+  height: 35px;
+  padding-left: 2rem;
+  background-color: var(--main-bright);
+  color: var(--main-dark);
+  border-radius: ${({ $isShown }) => ($isShown ? "20px" : "18px")};
+  font-size: 0.8rem;
+  border: 1px solid var(--main-dark);
+  transition: width 600ms linear;
+`;
+
+export default function SearchBar({ SetShownEntries, emotionEntries }) {
```

Comment:
The SearchBar is just the input element to interact with the user. It does not need any entries. Please import and use fuse.js in `pages/emotion-records.js` . That's where the list is rendered. There you can implement a handleSearch function which is send to the SearchBar as a prop and that is called in the onChange event.